src/packages/utils/src/utils/executor.ts

```diff
@@ -12,6 +12,15 @@ export class Executor {
     this.#requestCoordinator = requestCoordinator;
   }
 
+  public stop() {
```

Comment:
I'd rather not pass these functions through, but I figured this was the lesser of several evils, having considered:
- exposing the RequestCoordinator dependency from executor
- passing RequestCoordinator _and_ Executor as dependencies into Provider
- combining RequestCoordinator and Executor into a single dependency

Happy to hear opinions / alternatives.

Reply:
Alternatively, what if we:

 1. call/implement `executor.stop()`
    * which calls `requestCoordinator.stop()`
 2. call `await blockchain.stop()`
 3. call a new `executor.end()`
   * which calls a new `requestCoordinator.end()` which would do what `rejectPendingTasks()` does now.

I think this may be slightly different behavior from what you have now, but seems acceptable and the naming makes sense to me?

I may have missed some details, but it seems like a simple approach that isn't too dissimilar from what you have now.


---

I do think RequestCoordinator+Executor could use a small refactoring, but I'd rather not do that now.

Reply:
I've adopted your naming for simplicity :)